docker/licenses/extract-licenses.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+# Copyright 2021 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+append_license() {
+  lib=$1
+  path=$2
+  echo "================================================================================" >> ${TMP_LICENSES}
+  echo "= ${lib} licensed under: =" >> ${TMP_LICENSES}
+  echo >> ${TMP_LICENSES}
+  cat "$path" >> ${TMP_LICENSES}
+  echo >> ${TMP_LICENSES}
+  echo "= ${path} MD5 $(cat "${path}" | md5sum | awk '{print $1}')" >> ${TMP_LICENSES}
+  echo "================================================================================" >> ${TMP_LICENSES}
+  echo >> ${TMP_LICENSES}
+
+}
+
+SRC_ROOT=$(dirname "${BASH_SOURCE}")/../..
+TMP_LICENSES=${SRC_ROOT}/docker/licenses/LICENSE
+
+cd ${SRC_ROOT}
+
+# Please don't change this tmp folder, line 55 is hardcoded to this.
+TMP_FOLDER=/tmp/ggg
+
+echo "Collecting LICENCE files from go modules..."
+go-licenses save "src/go/configmanager/main/server.go" --save_path "${TMP_FOLDER}"
```

Comment:
Can we extract the script into two, one part for envoy, one part for configmanager, and trigger them under `make envoy-build` and `make build` targets so that they can be automatically added when we run test locally?  Also under presubmit test, we can check `git status` after integration test, to ensure the licenses are added in the PR?

Reply:
It seems that this is overkill.  License files do not change often. We only need to do this manually, when dependencies are updated.